### PR TITLE
Add server and client controlled cyclic step-by-steps

### PR DESCRIPTION
**Pull request by SYangster**
- server-controlled and client-controlled cyclic workflow step-by-steps
- created notebooks and jobs templates

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Comment by SYangster** [comment]
Will update client api fl code for multi-task for other examples in later PR

File: job_templates/cyclic_pt/config_fed_server.conf [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  format_version = 2
+  server {
+    heart_beat_timeout = 600
+  }
+  task_data_filters =[]
+  task_result_filters = []
+  # path to defined PyTorch network
+  model_class_path = "net.Net"
```
Comment (YuanTingHsieh) [inline job_templates/cyclic_pt/config_fed_server.conf:9]: suggest add:

https://github.com/NVIDIA/NVFlare/blob/main/job_templates/sag_pt/config_fed_server.conf#L11C1-L13C31

File: job_templates/cyclic_cc_pt/config_fed_client.conf [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  format_version = 2
+
+  # This is the application script which will be invoked. Client can replace this script with user's own training script.
+  app_script = "train.py"
+
+  # Additional arguments needed by the training code. For example, in lightning, these can be --trainer.batch_size=xxx.
+  app_config = ""
+
+  # path to defined PyTorch network
+  model_class_path = "net.Net"
```
Comment (YuanTingHsieh) [inline job_templates/cyclic_cc_pt/config_fed_client.conf:12]: suggest a bit more explanation as in https://github.com/NVIDIA/NVFlare/blob/main/job_templates/sag_pt/config_fed_server.conf#L11C1-L13C31

File: job_templates/cyclic_cc_pt/config_fed_client.conf [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  # Client Computing Executors.
+  executors = [
+    {
+      # tasks the executors are defined to handle
+      tasks = ["train"]
+
+      # This particular executor
+      executor {
+
+        # Eexecutor name : PTClientAPILauncherExecutor
```
Comment (YuanTingHsieh) [inline job_templates/cyclic_cc_pt/config_fed_client.conf:23]: typo: Executor name

File: job_templates/cyclic_cc_pt/config_fed_client.conf [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        }
+      }
+    }
+  ]
+
+  # this defined an array of task data filters. If provided, it will control the data from server controller to client executor
+  task_data_filters =  []
+
+  # this defined an array of task result filters. If provided, it will control the result from client executor to server controller
+  task_result_filters = []
```
Comment (YuanTingHsieh) [inline job_templates/cyclic_cc_pt/config_fed_client.conf:81]: This description is not 100% accurate in CCWF.

The task_data_filters filters things coming from Client side Controller -> Client side Executor right?
The task_result_filters filters things sending from Client side Executor -> Client side Controller?

@yhwen can you add more context in this?
We introduce the "direction" as well right?

Comment (SYangster) [reply job_templates/cyclic_cc_pt/config_fed_client.conf:81]: Oh right good catch, I forgot to update these comments from the previous template

Comment (SYangster) [reply job_templates/cyclic_cc_pt/config_fed_client.conf]: Updated comment for ccwf and filter direction (in, out, inout)

File: job_templates/cyclic_cc_pt/config_fed_client.conf [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+          learn_task_name = "train"
+          persistor_id = "persistor"
+          shareable_generator_id = "shareable_generator"
+        }
+      }
+    }
+  ]
+
+  # Array of task data filters. If provided, it will control the data from client controller to client executor
+  # Filter direction (in, out, inout) can be set as since clients send tasks to each other, a task has both a sending (out) and a receiving (in) direction
```
Comment (YuanTingHsieh) [inline job_templates/cyclic_cc_pt/config_fed_client.conf:79]: Thanks for the updates, we might need some documentation updates on this "filter" and direction and how they mean in client controlled workflows VS original server controlled workflows.

@nvkevlu @yhwen

**Review by YuanTingHsieh — COMMENTED** [review]
Great work!
I forgot, do we have some figures showcasing difference of server-controlled cyclic VS client-controlled cyclic workflow? (or more general: server-controlled workflow VS client-controlled workflow)

**Comment by SYangster** [comment]
> Great work! I forgot, do we have some figures showcasing difference of server-controlled cyclic VS client-controlled cyclic workflow? (or more general: server-controlled workflow VS client-controlled workflow)

Yes good idea. I had some figures for swarm learning, I can create something similar for server-controlled cyclic vs client-controlled cyclic

**Review by YuanTingHsieh — APPROVED** [review]
Thanks LGTM

**Pull request merged**